### PR TITLE
repair: reuse table name from repair_range argument

### PR DESCRIPTION
**Pull request by Deexie**
Currently in shard_repair_task_impl::repair_range table name is retrieved with database::find_column_family and in case of exception, we return from the function.

But the table name is already kept in table_info passed to repair_range as an argument. Let's reuse it. If a table is dropped, we will find it out almost immediately after calling repair_cf_range_row_level and handle it more adequately.

### CODE REVIEW

File: repair/repair.cc [hunk 1]
```diff
@@ -675,20 +675,15 @@ future<> repair::shard_repair_task_impl::repair_range(const dht::token_range& ra
     rlogger.debug("repair[{}]: Repair {} out of {} ranges, keyspace={}, table={}, range={}, peers={}, live_peers={}",
         global_repair_id.uuid(), ranges_index, ranges_size(), _status.keyspace, table.name, range, neighbors, live_neighbors);
     co_await mm.sync_schema(db.local(), neighbors);
-    sstring cf;
-    try {
-        cf = db.local().find_column_family(table.id).schema()->cf_name();
```
Comment (xemul) [inline repair/repair.cc:680]: This looks intentional -- up above there's call to sync_schema() that may drop that table

Comment (Deexie) [reply repair/repair.cc:680]: I believe it was not. `find_column_family` was called before table name could have been accessed from different source. Then try-catch clause was added as a fix for an issue (7f3eb8b4e8009337c0333a2966daeb43edb11d22) and the table name become accessible from a different place in 09abbdd. That order suggests it's kind of random.

Moreover with this try-catch we need to throw and handle the exception for each range in the table. Without it once `no_such_column_family` is thrown we mark it in `dropped_tables` (691) and then just skip it without throwing the exception (685) for other ranges.

Comment (Deexie) [reply repair/repair.cc:680]: @xemul please see

Comment (xemul) [reply repair/repair.cc:680]: OK, fair enough

**Pull request closed without merging**